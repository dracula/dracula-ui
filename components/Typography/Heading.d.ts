import React, { HTMLAttributes } from 'react';
import { SpacingPropType } from '../../base/spacing';
export declare const headingSizes: {
    'heading-1': string;
    'heading-2': string;
    'heading-3': string;
    'heading-4': string;
    'heading-5': string;
    'heading-6': string;
};
export declare const headingColors: {
    white: string;
    cyan: string;
    green: string;
    orange: string;
    pink: string;
    purple: string;
    red: string;
    yellow: string;
    purpleCyan: string;
    yellowPink: string;
    cyanGreen: string;
    pinkPurple: string;
    animated: string;
};
/** Heading Props */
export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
    /**
     * The size and type of Heading to be used.
     * Denotes hierarchy.
     */
    size?: keyof typeof headingSizes;
    /**
     * The Dracula UI color to be applied to the Heading.
     */
    color?: keyof typeof headingColors;
    /**
     * Controls the spacing between the Heading component and its parent and siblings.
     */
    spacing?: SpacingPropType;
}
/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export declare const Heading: React.FC<HeadingProps>;
