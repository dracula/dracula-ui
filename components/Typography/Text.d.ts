import React, { HTMLAttributes } from 'react';
import { SpacingPropType } from '../../base/spacing';
export declare const textSizes: {
    large: string;
    medium: string;
    small: string;
    xsmall: string;
};
export declare const textWeights: {
    normal: string;
    semibold: string;
    bold: string;
};
export declare const textColors: {
    white: string;
    cyan: string;
    green: string;
    orange: string;
    pink: string;
    purple: string;
    red: string;
    yellow: string;
    purpleCyan: string;
    yellowPink: string;
    cyanGreen: string;
    pinkPurple: string;
    animated: string;
};
/** Text Props */
export interface TextProps extends HTMLAttributes<HTMLSpanElement> {
    /**
     * Controls the size of the text based on pre-configured Dracula UI sizes.
     * Options: `xsmall`, `small`, `medium`, `large`
     */
    size?: keyof typeof textSizes;
    /**
     * Controsl the weight of the text.
     * Options: `normal`, `semibold`, `bold`.
     */
    weight?: keyof typeof textWeights;
    /**
     * Controls the color of the text
     */
    color?: keyof typeof textColors;
    /**
     * Controls the spacing between the Text component and its parent and siblings.
     */
    spacing?: SpacingPropType;
}
/**
 * Text is the base component for any sort of text.
 *
 * Consumers of this component can control, the color, size,
 * weight, and spacing of Text.
 *
 * Use this component for generic, and non-hierarchical text that is
 * to be displayed on a page, or as part of other complex components
 * or UI patterns
 */
export declare const Text: {
    (props: TextProps): JSX.Element;
    displayName: string;
};
/**
 * Paragraph is a semantic component used for blocks of text with
 * semantic meaning.
 *
 * Paragraph accepts all the same customization options as Text.
 */
export declare const Paragraph: React.FC<TextProps>;
