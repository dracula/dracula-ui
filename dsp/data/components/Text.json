{
  "class": "component",
  "type": "page",
  "id": "Text",
  "name": "Text",
  "last_updated": "2021-03-14T22:31:20.414Z",
  "last_updated_by": "System",
  "description": "Text is the base component for any sort of text.\n\nConsumers of this component can control, the color, size,\nweight, and spacing of Text.\n\nUse this component for generic, and non-hierarchical text that is\nto be displayed on a page, or as part of other complex components\nor UI patterns",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "text",
    "languages": {
      "javascriptreact": "<Text>Text</Text>;\n",
      "html": "<span class=\"drac-text drac-text-white\">Text</span>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<span class=\"drac-text drac-text-white\">Text</span>\n",
      "react": "<Text>Text</Text>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Text.png",
      "svgData": "./dsp/assets/svgs/Text.svg"
    },
    {
      "title": "Colors",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"white\" class=\"drac-text drac-text-white\">white</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"black\" class=\"drac-text drac-text-black\">black</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"blackSecondary\" class=\"drac-text drac-text-black-secondary\"\n      >blackSecondary</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"blackLight\" class=\"drac-text drac-text-black-light\"\n      >blackLight</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"cyan\" class=\"drac-text drac-text-cyan\">cyan</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"green\" class=\"drac-text drac-text-green\">green</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"orange\" class=\"drac-text drac-text-orange\">orange</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"pink\" class=\"drac-text drac-text-pink\">pink</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"purple\" class=\"drac-text drac-text-purple\">purple</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"red\" class=\"drac-text drac-text-red\">red</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"yellow\" class=\"drac-text drac-text-yellow\">yellow</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"purpleCyan\" class=\"drac-text drac-text-purple-cyan\"\n      >purpleCyan</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"yellowPink\" class=\"drac-text drac-text-yellow-pink\"\n      >yellowPink</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"cyanGreen\" class=\"drac-text drac-text-cyan-green\"\n      >cyanGreen</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"pinkPurple\" class=\"drac-text drac-text-pink-purple\"\n      >pinkPurple</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span color=\"animated\" class=\"drac-text drac-text-animated\">animated</span>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"white\">white</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"black\">black</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"blackSecondary\">blackSecondary</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"blackLight\">blackLight</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"cyan\">cyan</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"green\">green</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"orange\">orange</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"pink\">pink</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"purple\">purple</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"red\">red</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"yellow\">yellow</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"purpleCyan\">purpleCyan</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"yellowPink\">yellowPink</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"cyanGreen\">cyanGreen</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"pinkPurple\">pinkPurple</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text color=\"animated\">animated</Text>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/TextColors.png",
      "svgData": "./dsp/assets/svgs/TextColors.svg"
    },
    {
      "title": "Weights",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <span weight=\"normal\" class=\"drac-text drac-text-white\">normal</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span weight=\"semibold\" class=\"drac-text drac-text-semibold drac-text-white\"\n      >semibold</span\n    >\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span weight=\"bold\" class=\"drac-text drac-text-bold drac-text-white\"\n      >bold</span\n    >\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Text weight=\"normal\">normal</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text weight=\"semibold\">semibold</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text weight=\"bold\">bold</Text>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/TextWeights.png",
      "svgData": "./dsp/assets/svgs/TextWeights.svg"
    },
    {
      "title": "Sizes",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <span class=\"drac-text drac-text-lg drac-text-white\">large</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span class=\"drac-text drac-text-white\">medium</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span class=\"drac-text drac-text-sm drac-text-white\">small</span>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <span class=\"drac-text drac-text-xs drac-text-white\">xsmall</span>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Text size=\"large\">large</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text size=\"medium\">medium</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text size=\"small\">small</Text>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Text size=\"xsmall\">xsmall</Text>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/TextSizes.png",
      "svgData": "./dsp/assets/svgs/TextSizes.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "size": {
      "defaultValue": null,
      "description": "Controls the size of the text based on pre-configured Dracula UI sizes.\nOptions: `xsmall`, `small`, `medium`, `large`",
      "name": "size",
      "parent": {
        "fileName": "src/components/Text/Text.tsx",
        "name": "TextProps"
      },
      "declarations": [
        { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
      ],
      "required": false,
      "type": {
        "name": "\"large\" | \"medium\" | \"small\" | \"xsmall\" | undefined"
      }
    },
    "weight": {
      "defaultValue": null,
      "description": "Controsl the weight of the text.\nOptions: `normal`, `semibold`, `bold`.",
      "name": "weight",
      "parent": {
        "fileName": "src/components/Text/Text.tsx",
        "name": "TextProps"
      },
      "declarations": [
        { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
      ],
      "required": false,
      "type": { "name": "\"normal\" | \"semibold\" | \"bold\" | undefined" }
    },
    "color": {
      "defaultValue": null,
      "description": "Controls the color of the text",
      "name": "color",
      "parent": {
        "fileName": "src/components/Text/Text.tsx",
        "name": "TextProps"
      },
      "declarations": [
        { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Controls the spacing between the Text component and its parent and siblings.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Text/Text.tsx",
        "name": "TextProps"
      },
      "declarations": [
        { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "",
      "name": "as",
      "parent": {
        "fileName": "src/components/Text/Text.tsx",
        "name": "TextProps"
      },
      "declarations": [
        { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
      ],
      "required": false,
      "type": { "name": "\"a\" | \"span\" | \"p\" | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Text is the base component for any sort of text.\n\nConsumers of this component can control, the color, size,\nweight, and spacing of Text.\n\nUse this component for generic, and non-hierarchical text that is\nto be displayed on a page, or as part of other complex components\nor UI patterns",
    "displayName": "Text",
    "methods": [],
    "props": {
      "size": {
        "defaultValue": null,
        "description": "Controls the size of the text based on pre-configured Dracula UI sizes.\nOptions: `xsmall`, `small`, `medium`, `large`",
        "name": "size",
        "parent": {
          "fileName": "src/components/Text/Text.tsx",
          "name": "TextProps"
        },
        "declarations": [
          { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
        ],
        "required": false,
        "type": {
          "name": "\"large\" | \"medium\" | \"small\" | \"xsmall\" | undefined"
        }
      },
      "weight": {
        "defaultValue": null,
        "description": "Controsl the weight of the text.\nOptions: `normal`, `semibold`, `bold`.",
        "name": "weight",
        "parent": {
          "fileName": "src/components/Text/Text.tsx",
          "name": "TextProps"
        },
        "declarations": [
          { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
        ],
        "required": false,
        "type": { "name": "\"normal\" | \"semibold\" | \"bold\" | undefined" }
      },
      "color": {
        "defaultValue": null,
        "description": "Controls the color of the text",
        "name": "color",
        "parent": {
          "fileName": "src/components/Text/Text.tsx",
          "name": "TextProps"
        },
        "declarations": [
          { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Controls the spacing between the Text component and its parent and siblings.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Text/Text.tsx",
          "name": "TextProps"
        },
        "declarations": [
          { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "",
        "name": "as",
        "parent": {
          "fileName": "src/components/Text/Text.tsx",
          "name": "TextProps"
        },
        "declarations": [
          { "fileName": "src/components/Text/Text.tsx", "name": "TextProps" }
        ],
        "required": false,
        "type": { "name": "\"a\" | \"span\" | \"p\" | undefined" }
      }
    }
  }
}
