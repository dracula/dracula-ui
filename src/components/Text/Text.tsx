import cx from 'classnames/dedupe'
import { mapValues } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const textSizes = {
  md: 'drac-text',
  lg: 'drac-text-lg',
  sm: 'drac-text-sm',
  xs: 'drac-text-xs',
}

export const textWeights = {
  normal: 'drac-text',
  semibold: 'drac-text-semibold',
  bold: 'drac-text-bold'
}

export const lineHeights = {
  md: 'drac-line-height',
  xl: 'drac-line-height-xl',
  lg: 'drac-line-height-lg',
  sm: 'drac-line-height-sm',
  xs: 'drac-line-height-xs',
}

export const textColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Text Props */
export interface TextProps
  extends HTMLAttributes<HTMLSpanElement>,
    PaddingMixin,
    MarginMixin {
  /**
   * Controls the size of the text based on pre-configured Dracula UI sizes.
   */
  size?: keyof typeof textSizes

  /**
   * Controls the weight of the text.
   */
  weight?: keyof typeof textWeights

  /**
   * Controls the line height of a text.
   */
  lineHeight?: keyof typeof lineHeights

  /**
   * Controls the color of the text
   */
  color?: keyof typeof textColors

  /** The HTML element to be used */
  as?: 'a' | 'span' | 'p'
}

/**
 * Text is the base component for any sort of text.
 *
 * Consumers of this component can control, the color, size,
 * weight, and spacing of Text.
 *
 * Use this component for generic, and non-hierarchical text that is
 * to be displayed on a page, or as part of other complex components
 * or UI patterns
 */
export const Text = (props: TextProps) => {
  const { size, weight, lineHeight, as, color, ...htmlProps } = props

  const finalProps = {
    ...htmlProps,
    className: cx(
      `drac-text`,
      props.className,
      textSizes[size ?? 'md'],
      textWeights[weight ?? 'normal'],
      lineHeights[lineHeight ?? 'md'],
      textColors[color ?? 'white'],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return React.createElement(
    as ?? 'span',
    cleanProps(finalProps),
    props.children
  )
}

Text.displayName = 'Text'
