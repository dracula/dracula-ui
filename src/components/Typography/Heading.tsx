import cx from 'classnames'
import { mapValues } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import { spacingClasses, SpacingPropType } from '../../base/spacing'

export const headingSizes = {
  'heading-1': 'drac-heading-1',
  'heading-2': 'drac-heading-2',
  'heading-3': 'drac-heading-3',
  'heading-4': 'drac-heading-4',
  'heading-5': 'drac-heading-5',
  'heading-6': 'drac-heading-6'
}

export const headingColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Heading Props */
export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
  /**
   * The size and type of Heading to be used.
   * Denotes hierarchy.
   */
  size?: keyof typeof headingSizes

  /**
   * The Dracula UI color to be applied to the Heading.
   */
  color?: keyof typeof headingColors

  /**
   * Controls the spacing between the Heading component and its parent and siblings.
   */
  spacing?: SpacingPropType
}

/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export const Heading: React.FC<HeadingProps> = (props: HeadingProps) => {
  const tag = {
    'heading-1': 'h1',
    'heading-2': 'h2',
    'heading-3': 'h3',
    'heading-4': 'h4',
    'heading-5': 'h5',
    'heading-6': 'h6'
  }

  const size = tag[props.size ?? 'heading-1']
  const finalProps = {
    ...props,
    className: cx(
      `drac-heading`,
      headingSizes[props.size ?? 'heading-1'],
      headingColors[props.color ?? 'white'],
      ...spacingClasses(props.spacing ?? 'none')
    )
  }

  return React.createElement(size, finalProps, props.children)
}

Heading.displayName = 'Heading'
