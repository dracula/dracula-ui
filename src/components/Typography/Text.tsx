import React, { HTMLAttributes } from 'react'
import { mapValues } from 'lodash'
import cx from 'classnames'
import { spacingClasses, SpacingPropType } from '../../base/spacing'
import { baseTextColors } from '../../base/colors'

export const textSizes = {
  large: 'drac-text-lg',
  medium: 'drac-text',
  small: 'drac-text-sm',
  xsmall: 'drac-text-xs'
}

export const textWeights = {
  normal: 'drac-text',
  semibold: 'drac-text-semibold',
  bold: 'drac-text-bold'
}

export const textColors = mapValues(baseTextColors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Text Props */
export interface TextProps extends HTMLAttributes<HTMLSpanElement> {
  /**
   * Controls the size of the text based on pre-configured Dracula UI sizes.
   * Options: `xsmall`, `small`, `medium`, `large`
   */
  size?: keyof typeof textSizes

  /**
   * Controsl the weight of the text.
   * Options: `normal`, `semibold`, `bold`.
   */
  weight?: keyof typeof textWeights

  /**
   * Controls the color of the text
   */
  color?: keyof typeof textColors

  /**
   * Controls the spacing between the Text component and its parent and siblings.
   */
  spacing?: SpacingPropType
}

/**
 * Text is the base component for any sort of text.
 *
 * Consumers of this component can control, the color, size,
 * weight, and spacing of Text.
 *
 * Use this component for generic, and non-hierarchical text that is
 * to be displayed on a page, or as part of other complex components
 * or UI patterns
 */
export const Text = (props: TextProps) => {
  const finalProps = {
    ...props,

    className: cx(
      `drac-text`,
      textSizes[props.size ?? 'medium'],
      textWeights[props.weight ?? 'normal'],
      textColors[props.color ?? 'white'],
      spacingClasses(props.spacing)
    )
  }

  return <span {...finalProps}>{props.children}</span>
}

/**
 * Paragraph is a semantic component used for blocks of text with
 * semantic meaning.
 *
 * Paragraph accepts all the same customization options as Text.
 */
export const Paragraph: React.FC<TextProps> = (props: TextProps) => {
  const finalProps = {
    ...props,

    className: cx(
      `drac-text`,
      textSizes[props.size ?? 'medium'],
      textWeights[props.weight ?? 'normal'],
      textColors[props.color ?? 'white'],
      spacingClasses(props.spacing ?? 'smallY')
    )
  }

  return <p {...finalProps}>{props.children}</p>
}

Text.displayName = 'Text'
