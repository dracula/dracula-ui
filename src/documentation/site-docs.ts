import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

jest.setTimeout(10000)

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name
  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeEach(async () => {
      browser = await puppeteer.launch({ headless: true, devtools: true })
    })

    afterEach(async () => {
      await browser.close()
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      const [screenshot, svg] = await componentScreenshot(
        browser,
        snapshot,
        null,
        name
      )

      snapshot.screenshot = screenshot
      snapshot.svgData = svg

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          variation,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
