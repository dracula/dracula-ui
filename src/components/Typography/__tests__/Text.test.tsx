import { Text } from '@/components/Typography/Text'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/documentation/site-docs'
import {
  TextSizes,
  TextBasic,
  TextWeights,
  TextColors
} from '@/components/Typography/__stories__/Text.stories'

siteDocs(Text, {
  basicUsage() {
    return buildSnapshot('Usage', TextBasic)
  },
  variations() {
    return [
      buildSnapshot('Colors', TextColors),
      buildSnapshot('Weights', TextWeights),
      buildSnapshot('Sizes', TextSizes)
    ] as SnapshotBuilder[]
  }
})
