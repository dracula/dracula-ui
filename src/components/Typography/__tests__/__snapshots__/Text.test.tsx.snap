// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Text Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<span class=\\"drac-text drac-text drac-text drac-text-white\\">Text</span>
",
  "react": "<Text>Text</Text>;
",
  "title": "Usage",
}
`;

exports[`Site: Text Colors 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"white\\" class=\\"drac-text drac-text drac-text drac-text-white\\"
      >white</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"cyan\\" class=\\"drac-text drac-text drac-text drac-text-cyan\\"
      >cyan</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"green\\" class=\\"drac-text drac-text drac-text drac-text-green\\"
      >green</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"orange\\" class=\\"drac-text drac-text drac-text drac-text-orange\\"
      >orange</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"pink\\" class=\\"drac-text drac-text drac-text drac-text-pink\\"
      >pink</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"purple\\" class=\\"drac-text drac-text drac-text drac-text-purple\\"
      >purple</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"red\\" class=\\"drac-text drac-text drac-text drac-text-red\\"
      >red</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span color=\\"yellow\\" class=\\"drac-text drac-text drac-text drac-text-yellow\\"
      >yellow</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      color=\\"purpleCyan\\"
      class=\\"drac-text drac-text drac-text drac-text-purple-cyan\\"
      >purpleCyan</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      color=\\"yellowPink\\"
      class=\\"drac-text drac-text drac-text drac-text-yellow-pink\\"
      >yellowPink</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      color=\\"cyanGreen\\"
      class=\\"drac-text drac-text drac-text drac-text-cyan-green\\"
      >cyanGreen</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      color=\\"pinkPurple\\"
      class=\\"drac-text drac-text drac-text drac-text-pink-purple\\"
      >pinkPurple</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      color=\\"animated\\"
      class=\\"drac-text drac-text drac-text drac-text-animated\\"
      >animated</span
    >
  </div>
</div>
",
  "react": "<div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"white\\">white</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"cyan\\">cyan</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"green\\">green</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"orange\\">orange</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"pink\\">pink</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"purple\\">purple</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"red\\">red</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"yellow\\">yellow</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"purpleCyan\\">purpleCyan</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"yellowPink\\">yellowPink</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"cyanGreen\\">cyanGreen</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"pinkPurple\\">pinkPurple</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text color=\\"animated\\">animated</Text>
  </div>
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: Text Sizes 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <div style=\\"margin-bottom: 20px\\">
    <span class=\\"drac-text drac-text-lg drac-text drac-text-white\\">large</span>
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span class=\\"drac-text drac-text drac-text drac-text-white\\">medium</span>
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span class=\\"drac-text drac-text-sm drac-text drac-text-white\\">small</span>
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span class=\\"drac-text drac-text-xs drac-text drac-text-white\\">xsmall</span>
  </div>
</div>
",
  "react": "<div>
  <div style={{ marginBottom: 20 }}>
    <Text size=\\"large\\">large</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text size=\\"medium\\">medium</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text size=\\"small\\">small</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text size=\\"xsmall\\">xsmall</Text>
  </div>
</div>;
",
  "title": "Sizes",
}
`;

exports[`Site: Text Weights 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <div style=\\"margin-bottom: 20px\\">
    <span weight=\\"normal\\" class=\\"drac-text drac-text drac-text drac-text-white\\"
      >normal</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      weight=\\"semibold\\"
      class=\\"drac-text drac-text drac-text-semibold drac-text-white\\"
      >semibold</span
    >
  </div>
  <div style=\\"margin-bottom: 20px\\">
    <span
      weight=\\"bold\\"
      class=\\"drac-text drac-text drac-text-bold drac-text-white\\"
      >bold</span
    >
  </div>
</div>
",
  "react": "<div>
  <div style={{ marginBottom: 20 }}>
    <Text weight=\\"normal\\">normal</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text weight=\\"semibold\\">semibold</Text>
  </div>
  <div style={{ marginBottom: 20 }}>
    <Text weight=\\"bold\\">bold</Text>
  </div>
</div>;
",
  "title": "Weights",
}
`;
